modules/aws_account_permissions/main.tf

```
@@ -0,0 +1,208 @@
+
+data "aws_partition" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+resource "random_uuid" "role_secret" {}
+
+resource "aws_iam_role_policy" "IllumioCloudAWSIntegrationPolicy" {
+  name = "IllumioCloudAWSIntegrationPolicy"
+  role = aws_iam_role.illumio_cloud_integration_role.id
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = [
+          "apigateway:GET",
+          "autoscaling:Describe*",
+          "cloudtrail:DescribeTrails",
+          "cloudtrail:GetTrailStatus",
+          "cloudtrail:LookupEvents",
+          "cloudwatch:Describe*",
+          "cloudwatch:Get*",
+          "cloudwatch:List*",
+          "codedeploy:List*",
+          "codedeploy:BatchGet*",
+          "directconnect:Describe*",
+          "docdb-elastic:GetCluster",
+          "docdb-elastic:ListTagsForResource",
+          "dynamodb:List*",
+          "dynamodb:Describe*",
+          "ec2:Describe*",
+          "ec2:SearchTransitGatewayMulticastGroups",
+          "ecs:Describe*",
+          "ecs:List*",
+          "elasticache:Describe*",
+          "elasticache:List*",
+          "elasticfilesystem:DescribeAccessPoints",
+          "elasticfilesystem:DescribeFileSystems",
+          "elasticfilesystem:DescribeTags",
+          "elasticloadbalancing:Describe*",
+          "elasticmapreduce:List*",
+          "elasticmapreduce:Describe*",
+          "es:ListTags",
+          "es:ListDomainNames",
+          "es:DescribeElasticsearchDomains",
+          "fsx:DescribeFileSystems",
+          "fsx:ListTagsForResource",
+          "health:DescribeEvents",
+          "health:DescribeEventDetails",
+          "health:DescribeAffectedEntities",
+          "kinesis:List*",
+          "kinesis:Describe*",
+          "lambda:GetPolicy",
+          "lambda:List*",
+          "logs:TestMetricFilter",
+          "logs:DescribeSubscriptionFilters",
+          "organizations:Describe*",
+          "organizations:List*",
+          "rds:Describe*",
+          "rds:List*",
+          "redshift:DescribeClusters",
+          "redshift:DescribeLoggingStatus",
+          "route53:List*",
+          "s3:GetBucketLogging",
+          "s3:GetBucketLocation",
+          "s3:GetBucketNotification",
+          "s3:GetBucketTagging",
+          "s3:ListAllMyBuckets",
+          "sns:List*",
+          "sqs:ListQueues",
+          "states:ListStateMachines",
+          "states:DescribeStateMachine",
+          "support:DescribeTrustedAdvisor*",
+          "support:RefreshTrustedAdvisorCheck",
+          "tag:GetResources",
+          "tag:GetTagKeys",
+          "tag:GetTagValues",
+          "xray:BatchGetTraces",
+          "xray:GetTraceSummaries"
+        ],
+        Resource = "*"
+      }
+    ]
+  })
+}
+resource "aws_iam_role_policy" "IllumioCloudAWSProtectionPolicy" {
+  count = var.access_mode == "ReadWrite" ? 1 : 0
+  name = "IllumioCloudAWSProtectionPolicy"
```

1. Don't hardcode names.
2. Define tags.
```suggestion
  name = var.iam_name_prefix + "ProtectionPolicy"
  tags = var.tags
```

---

I fixed this.